akka-actor/src/main/scala/akka/actor/AbstractProps.scala

```diff
@@ -59,6 +60,19 @@ private[akka] trait AbstractProps {
     create(classOf[CreatorConsumer], actorClass, creator)
   }
 
+  private final def isNotClosingOverEnclosingClass(clazz: Class[_]): Boolean = {
+    def hasDeclaredConstructorWithEmptyParams: Boolean =
+      try { clazz.getDeclaredConstructor(); true }
+      catch { case ex: NoSuchMethodException ⇒ false }
+
+    def checkConstructors = {
+      val constructorsLength = clazz.getConstructors.length
+      constructorsLength > 0 || (constructorsLength == 0 && hasDeclaredConstructorWithEmptyParams)
+    }
+
+    (clazz.getEnclosingClass eq null) || checkConstructors
+  }
```

Comment:
I changed this part, compare with https://github.com/akka/akka/pull/17412/files#r30890616